src/components/ArticleItem.jsx

```diff
@@ -1,10 +1,11 @@
 import React from "react";
 import { Segment, Grid } from "semantic-ui-react";
+import { Article } from "../modules/apiHelper";
 
 const ArticleItem = ({ articleItem, id }) => {
   return (
     <Grid.Column width="4" data-cy="news-section">
-      <Segment data-cy={`article-${id}`}>
+      <Segment data-cy={`article-${id}`} onClick={() => Article.show(id)}> 
```

Comment:
I dont like your implementation. What I can see is that you have a bunch of onClick events like that one that call for Article.show or Artilce.index functions. I would say this is not the way it is usually done because it is spagettiish. Look at it from my side, I see this onClick event and wonder what does it do... then I look at Article.show function and see that it save response in Redux.... and that's it.... I have no clue what does that change in application... no way to know what is triggered where.

Another problem is, what if I want to share this article with someone...URL I copy will lead to the main page... so articles are not shareable

Reply:
You should have one route for list of articles, and one route for individual article. And you need to replace those onClick requests for Article to redirects to new route. And requests for Articles should happen on initial render when you change the route

Reply:
We have changed the implementation to use routes, setting the url path using application state. Here is an outline of the flow:
- `HomePage` renders and calls the `Article.index` method from the `apiHelper` module
- `HomePage` stores the returned articles and maps them onto instances of the `ArticleItem` component
- User clicks on an article title/lede and in so doing, the `ArticleItem` component stores the article `id` in application state
- `App.jsx` creates the route based on the article `id` stored in application state
- The `IndividualArticle` component calls `Article.show(id)`  when it renders and displays the article

Reply:
Disregard the above, we now rely on `react-router`'s built in hook `useParams` to set the URL and the path.